### PR TITLE
REOPENED - Integrate Revival Protocol Fixes in CHL #1237 

### LINKED ISSUE #1235: Revival Protocol restores action points to disoriented units
The blame is with the Revival Protocol blindly applying `X2Effect_RestoreActionPoints` to all targets.

The effect itself is poorly coded as well:
```cs
simulated protected function OnEffectAdded(const out EffectAppliedData ApplyEffectParameters, XComGameState_BaseObject kNewTargetState, XComGameState NewGameState, XComGameState_Effect NewEffectState)
{
	local XComGameState_Unit TargetUnit;

	TargetUnit = XComGameState_Unit(kNewTargetState);
	if (TargetUnit != none)
	{
		while (TargetUnit.NumActionPoints(class'X2CharacterTemplateManager'.default.StandardActionPoint) < class'X2CharacterTemplateManager'.default.StandardActionsPerTurn)
		{
			TargetUnit.ActionPoints.AddItem(class'X2CharacterTemplateManager'.default.StandardActionPoint);
		}
	}
}
```
Arguably it should call `TargetUnit.GiveStandardActionPoints()` instead.

This bug and others is addressed by RM's [[WotC] Revival Protocol Fixes](https://steamcommunity.com/sharedfiles/filedetails/?id=1123037187).

### PR DESCRIPTION
**Pull request by BlackDog86**
Fixes #1235

Reopening accidentally closed request.

Adds custom X2Condition to the RestoreActionPoints effect of Revival Protocol to prevent it from restoring APs to disoriented units. Updated the ability targetting condition to allow revival protocol to target stunned units, remove the stunned status and restore APs properly. Also integrated the additional targeting condition from RMs mod which allows the ability to target resistance soldiers. Discussion to be had over whether this part is required/necessary. 

Still requires finessing & testing before merge.

### CODE REVIEW

**Comment by Iridar** [comment]
> allows the ability to target resistance soldiers. Discussion to be had over whether this part is required/necessary.

In cases like this I would defer to in-game localization and reasonable player expectations. 

Localization goes:

> "Send the GREMLIN to an ally to remove any negative mental status effects: Disoriented, Stunned, Panicked, or Unconscious."

"an ally" would reasonably include any allied unit, including resistance soldiers.

The ability itself engages the GREMLIN to perform actions on a unit, requiring no special training from the unit, other than being human or close enough to a human (Skirmishers), so a reasonable person would expect it to be able to target resistance soldiers.

So I'd say yeah, valid change.

**Comment by Iridar** [comment]
This PR seems to be missing the .x2proj change with addition of new files.

**Comment by BlackDog86** [comment]
> This PR seems to be missing the .x2proj change with addition of new files.

Resolved :)

**Comment by Iridar** [comment]
Sorry, but it needs to be in a separate commit.

**Comment by BlackDog86** [comment]
That should do it

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/X2Ability_SpecialistAbilitySet.uc [hunk 1]
```diff
@@ -740,9 +742,18 @@ static function X2AbilityTemplate RevivalProtocol()
...
 	Template.AbilityTargetConditions.AddItem(new class'X2Condition_RevivalProtocol');
 
-	Template.AddTargetEffect(RemoveAdditionalEffectsForRevivalProtocolAndRestorativeMist());
-	Template.AddTargetEffect(new class'X2Effect_RestoreActionPoints');      //  put the unit back to full actions
-
+	Template.AddTargetEffect(RemoveAdditionalEffectsForRevivalProtocolAndRestorativeMist());	
+
+	// Start Issue #1235
+	/// HL-Docs: ref:Bugfixes; issue:1235
+	/// Update the ability template to call recover from stun & add a new condition to the Restore Action Point effect
```
Comment (Iridar) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/X2Ability_SpecialistAbilitySet.uc:749]: Sorry, but I have to nitpick to these docs, they need to be user-facing and explain what specific bugs are getting fixed in plain terms.

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/X2Ability_SpecialistAbilitySet.uc [hunk 2]
```diff
@@ -740,9 +742,18 @@ static function X2AbilityTemplate RevivalProtocol()
...
+	Template.AddTargetEffect(class'X2StatusEffects'.static.CreateStunRecoverEffect());
+	/// Restore APs on the unit only if the new X2Condition is met
```
Comment (Iridar) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/X2Ability_SpecialistAbilitySet.uc:751]: This comment should be a simple double `//`

**Review by Iridar — CHANGES_REQUESTED** [review]

**Comment by BlackDog86** [comment]
Updated the docs to be more descriptive and removed the errant /

**Comment by Iridar** [comment]
Closing this in favor of #1415, which adds more bugfixes.

**Pull request closed without merging**